includes/database/queries/class-adjustment.php

```diff
@@ -167,4 +167,75 @@ class Adjustment extends Query {
 	public function __construct( $query = array() ) {
 		parent::__construct( $query );
 	}
+
+	/**
+	 * Handles adding an adjustment to the database.
+	 * Tax rates are unique in that only one can be active for one unique country/region combination,
+	 * so if the adjustment is a tax rate, we need to do some extra checking.
+	 *
+	 * @param array $data
+	 * @return int|false
+	 */
+	public function add_item( $data = array() ) {
```

Comment:
This whole section feels really awkward and off. We're in the `class-adjustment.php`, yet we're calling things by `$existing_tax_rate_id`. I know the goal is to not allow duplicates, but we also shouldn't be 'updating' tax rates. Those are intended to be immutable, yet if we detect an existing tax rate we'll be able to modify it with the adjustment class. We need strong check around this `item_call` that we check the type before we determine if we allow updating or adding, and we should rename the methods to be more about being adjustments in general, and call out types of adjustments needing different treatments with comments.

Reply:
That's fair, and merits discussion. The reason the logic is in here to potentially update a tax rate is because that's what is currently in the tax compatibility class (if `update_option` is used to add tax rates). So I worked through this here to be consistent with that behavior. I agree, though, it seems off.

But the question is, then, is that wrong? And if so, how should this be handled? Should a new tax rate be registered as active and the original marked as inactive? Or should it silently fail?

If the former behavior is what it should be, then we can look at `edd_maybe_add_customer_address` as a starting point, and pull all of this logic out of the adjustment class and into a new `edd_add_tax_rate` function, and any time a tax rate is added, run it through that instead of directly through `edd_add_adjustment`. That would have some advantages in that we could ensure that tax rates are always complete (I found, for example, that I could add a tax rate with missing parameters--I think the status was one, and possibly even type?).

Reply:
Currently working to update this using a new `edd_add_tax_rate` function; I think that approach makes the most sense. It would still be possible to get into trouble using `edd_add_adjustment` directly but we can treat tax rates as specific kinds of objects.

Reply:
@robincornett I think taking our queues from the `edd_maybe_add_customer_address` is a decent avenue to pursue.